crates/uv-pypi-types/src/dependency_groups.rs

```diff
@@ -0,0 +1,148 @@
+use serde::{Deserialize, Deserializer};
+use std::collections::BTreeMap;
+use std::str::FromStr;
+
+use uv_normalize::GroupName;
+
+#[derive(Debug, Clone, PartialEq, serde::Serialize)]
+pub struct DependencyGroups(BTreeMap<GroupName, Vec<DependencyGroupSpecifier>>);
```

Comment:
Did you consider putting this file-move into a separate PR? It looks separable from the rest of the changes, and would be easy to merge on its own.

Reply:
That's true. Would you rather I create that PR now, or just for the future?

Reply:
It's up to you. I think it's easier to do it even in this case -- we can merge that PR immediately, it makes this one easier to review, etc.

Reply:
Done. The other PR has been merged and this one simplified